ServerCore/Pages/Puzzles/FileManagement.cshtml.cs

```diff
@@ -287,6 +287,36 @@ private bool PuzzleExists(int puzzleId)
             return _context.Puzzles.Any(e => e.ID == puzzleId);
         }
 
+        /// <summary>
+        /// Promotes files named "index.html" to the CustomURL or CustomSolutionURL properties of the puzzle.
+        /// </summary>
+        /// <param name="file">The file being uploaded</param>
+        private void UpdateCustomURLs(ContentFile file)
+        {
+            // Skip this if:
+            // - the file is not named index.html
+            // - the file is not a material or solution file
+            // - the appropriate Custom[Solution]URL is non-blank
+            if (!file.ShortName.EndsWith("index.html") ||
+                !(file.FileType == ContentFileType.PuzzleMaterial || file.FileType == ContentFileType.SolveToken) || 
+                (file.FileType == ContentFileType.PuzzleMaterial && !string.IsNullOrEmpty(Puzzle.CustomURL)) ||
+                (file.FileType == ContentFileType.SolveToken && !string.IsNullOrEmpty(Puzzle.CustomSolutionURL)))
+            {
+                return;
+            }
+
+            // lack of translation of {eventId} is very intentional - this is translated at runtime and makes the field value portable
+            string filePath = "/{eventId}/Files/" + file.ShortName;
```

Comment:
Path.Combine would be better practice here

Reply:
Except it adds the wrong slash...